streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingJobProgressListener.scala

```diff
@@ -40,6 +43,8 @@ private[streaming] class StreamingJobProgressListener(ssc: StreamingContext)
   private var totalProcessedRecords = 0L
   private val receiverInfos = new HashMap[Int, ReceiverInfo]
 
+  private val batchTimeToBatchUIData = new HashMap[Time, BatchUIData]
```

Comment:
This is good! But still not enough. The whole point of adding BatchUIData to encapsulate BatchInfo and outputOpId --> SparkJobId in it. So that all of that can be cleaned easily and we dont have to maintain separate data structures of BatchInfos (that is, waitingBatchInfos, runningBatchInfos, etc) and BatchUIData. So we should convert all (waiting/running/completed)BatchInfos to ***BatchUIData. Then we dont need to add this extra `batchTimeToBatchUIData` any more. And we dont need two methods - getBatchInfo and getBatchUIData, only the latter is needed.